types/react-native-listener/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// Type definitions for react-native-listener 1.1
+// Project: https://github.com/erikras/react-native-listener
+// Definitions by: Chintan <https://github.com/ChintanAcharya>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { Component, ReactElement } from 'react';
+
+type NativeEvent =
+    | 'Click'
+    | 'ContextMenu'
+    | 'DoubleClick'
+    | 'Drag'
+    | 'DragEnd'
+    | 'DragEnter'
+    | 'DragExit'
+    | 'DragLeave'
+    | 'DragOver'
+    | 'DragStart'
+    | 'Drop'
+    | 'Focus'
+    | 'KeyDown'
+    | 'KeyPress'
+    | 'KeyUp'
+    | 'MouseDown'
+    | 'MouseEnter'
+    | 'MouseLeave'
+    | 'MouseMove'
+    | 'MouseOut'
+    | 'MouseOver'
+    | 'MouseUp'
+    | 'PointerCancel'
+    | 'PointerDown'
+    | 'PointerEnter'
+    | 'PointerLeave'
+    | 'PointerMove'
+    | 'PointerOut'
+    | 'PointerOver'
+    | 'PointerUp'
+    | 'Scroll'
+    | 'TouchCancel'
+    | 'TouchEnd'
+    | 'TouchMove'
+    | 'TouchStart';
+
+type ExtendedWindowEventMap = {
+    doubleclick: WindowEventMap['dblclick'];
+    dragexit: WindowEventMap['dragleave'];
+} & WindowEventMap;
+
+type NativeListenerBubbleProps = {
+    [EventName in NativeEvent as `on${EventName}`]?: (event: ExtendedWindowEventMap[Lowercase<EventName>]) => any;
```

Comment:
The return type for the listener function is defined as `any` in the [`addEventListener` declaration](https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L25701). 

This package [uses `addEventListener` internally](https://github.com/erikras/react-native-listener/blob/master/src/NativeListener.js#L81), which is why the return type for the prop is also `any`.